File: auth-authorization-server-api/src/main/java/com/cartobucket/auth/authorization/server/routes/mappers/AccessTokenResponseMapper.java
package com.cartobucket.auth.authorization.server.routes.mappers;

import com.cartobucket.auth.data.domain.AccessToken;
import com.cartobucket.auth.authorization.server.dto.AccessTokenResponse;

public class AccessTokenResponseMapper {

    public static AccessTokenResponse toAccessTokenResponse(AccessToken accessToken) {
        var accessTokenResponse = new AccessTokenResponse();
        accessTokenResponse.setAccessToken(accessToken.getAccessToken());
        accessTokenResponse.setExpiresIn(accessToken.getExpiresIn());
        // Convert domain enum to DTO enum using the string value
        if (accessToken.getTokenType() != null) {
            accessTokenResponse.setTokenType(
                AccessTokenResponse.TokenTypeEnum.fromValue(accessToken.getTokenType().value())
            );
        }
        
        // Only set optional fields if they have values
        if (accessToken.getScope() != null) {
            accessTokenResponse.setScope(accessToken.getScope());
        }
        if (accessToken.getRefreshToken() != null) {
            accessTokenResponse.setRefreshToken(accessToken.getRefreshToken());
        }
        if (accessToken.getIdToken() != null) {
            accessTokenResponse.setIdToken(accessToken.getIdToken());
        }
        
        return accessTokenResponse;
    }
}

File: auth-authorization-server-api/src/main/kotlin/com/cartobucket/auth/authorization/server/routes/mappers/AccessTokenResponseMapper.kt
// (C)2024
package com.cartobucket.auth.authorization.server.routes.mappers

import com.cartobucket.auth.authorization.server.dto.AccessTokenResponse
import com.cartobucket.auth.data.domain.AccessToken

object AccessTokenResponseMapper {
    fun AccessToken.toAccessTokenResponse(): AccessTokenResponse =
        AccessTokenResponse(
            accessToken = this.accessToken,
            expiresIn = this.expiresIn,
            tokenType =
                this.tokenType?.let {
                    AccessTokenResponse.TokenTypeEnum.fromValue(it.value())
                },
            scope = this.scope,
            refreshToken = this.refreshToken,
            idToken = this.idToken,
        )
}
